app/components/Navbar/NavBarLink.tsx

```diff
@@ -0,0 +1,40 @@
+"use client";
+
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import styles from './NavBarLink.module.css';
+
+interface NavBarLinkProps {
+  title: string;
+  href: string;
+  links?: { name: string, href: string }[];  // Optional dropdown links
+}
+
+const NavBarLink: React.FC<NavBarLinkProps> = ({ title, href, links }) => {
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const toggleDropdown = () => {
+    setDropdownOpen(!dropdownOpen);
+  };
+
+  return (
+    <li className={styles.navItem} onClick={links ? toggleDropdown : undefined}>
```

Comment:
The dropdown only closes when you click on the icon again. It should also close when the cursor moves away from the menu without selecting.